src/mon/PGMap.h

```diff
@@ -147,6 +147,43 @@ class PGMapDigest {
   void pool_cache_io_rate_summary(Formatter *f, ostream *out,
                                   uint64_t poolid) const;
 
+  /**
+   * Return the number of additional bytes that can be stored in this
+   * pool before the first OSD fills up, accounting for PG overhead.
+   */
+  int64_t get_pool_free_space(const OSDMap &osd_map, int poolid) const;
+
+  float pool_raw_used_rate(const OSDMap &osd_map, int poolid) const
```

Comment:
Some of these stats gathering functions are rather new; we seem to be interested in exposing more statistics that rely on telling the user how many more bytes they're allowed to store.

 It's certainly only used in PGMap.cc for now, though, so I'll move it there.